src/domains/domains.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef DOMAINS_H_
+#define DOMAINS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CONFIG_SOC_FAMILY_RCAR)
+#include <xen_dom_mgmt.h>
+
+extern struct xen_domain_cfg domd_cfg;
+
+inline int create_domains(void)
+{
+	return domain_create(&domd_cfg, 1);
+};
+
+#else
+#define create_domains() 0
+
+#if !defined(CONFIG_NATIVE_APPLICATION)
+#include <domain.h>
```

Comment:
It is special case to xenvm. What about create domd_cfg_xenvm.c with empty config inside? And link this file for xenvm?

Reply:
Thanks, Made special case for xenvm